test/common/pktio_common.h

```diff
@@ -0,0 +1,124 @@
+extern "C" {
+#endif
+
+/**
+ * Print pktio link info
+ *
+ * @param pktio        Pktio handle
+ * @param pktio_name   Pktio name
+ */
+static void pktio_common_print_link_info(odp_pktio_t pktio, const char *pktio_name)
+{
+	odp_pktio_link_info_t info;
+
+	if (odp_pktio_link_info(pktio, &info)) {
+		ODPH_ERR("Error (%s): Printing pktio link info not possible, "
+			 "information retrieval failed.\n",
+			 pktio_name);
+		return;
+	}
+
+	printf("pktio: %s\n", pktio_name);
+	printf("  autoneg       %s\n",
+	       (info.autoneg == ODP_PKTIO_LINK_AUTONEG_ON ? "on" :
+	       (info.autoneg == ODP_PKTIO_LINK_AUTONEG_OFF ? "off" : "unknown")));
+	printf("  duplex        %s\n",
+	       (info.duplex == ODP_PKTIO_LINK_DUPLEX_HALF ? "half" :
+	       (info.duplex == ODP_PKTIO_LINK_DUPLEX_FULL ? "full" : "unknown")));
+	printf("  media         %s\n", info.media);
+	printf("  pause_rx      %s\n",
+	       (info.pause_rx == ODP_PKTIO_LINK_PAUSE_ON ? "on" :
+	       (info.pause_rx == ODP_PKTIO_LINK_PAUSE_OFF ? "off" : "unknown")));
+	printf("  pause_tx      %s\n",
+	       (info.pause_tx == ODP_PKTIO_LINK_PAUSE_ON ? "on" :
+	       (info.pause_tx == ODP_PKTIO_LINK_PAUSE_OFF ? "off" : "unknown")));
+	printf("  speed(Mbit/s) %" PRIu32 "\n\n", info.speed);
+}
+
+/**
+ * Print multiple pktio link infos.
+ * Same as pktio_common_print_link_info() but takes arrays of pktio handles and names
+ * and prints info for each link.
+ * 
+ * @param pktios        Pktio handles
+ * @param pktio_names   Pktio names
+ * @param num_pktio     Number of pktio handles and names
+ */
+static void pktio_common_print_link_info_multi(odp_pktio_t pktios[],
+					       const char *pktio_names[],
+					       uint32_t num_pktio)
+{
+	for (uint32_t i = 0; i < num_pktio; i++)
+		pktio_common_print_link_info(pktios[i], pktio_names[i]);
+}
+
+/**
+ * Wait for pktio links to be up and return status
+ *
+ * @param pktios        Pktio handles
+ * @param pktio_names   Pktio names
+ * @param num_pktio     Number of pktio handles and names
+ * @param wait_sec      Seconds to wait for all links to be up
+ *
+ * @retval 0 on success
+ * @retval -1 on failure
+ */
+static int pktio_common_check_link_status_wait(odp_pktio_t pktios[],
+					      const char *pktio_names[],
+					      uint32_t num_pktio,
+					      uint32_t wait_sec)
+{
+	odp_pktio_t pktio;
+	const char *name;
+	int link_status;
+	uint32_t link_wait = 0; /**< Centiseconds waited */
+
+	for (uint32_t i = 0; wait_sec && i < num_pktio; i++) {
+		while (1) {
+			pktio = pktios[i];
+			name = pktio_names[i];
+			link_status = odp_pktio_link_status(pktio);
+
+			if (link_status == ODP_PKTIO_LINK_STATUS_UP)
+				break;
+
+			link_wait++;
+
+			if (link_wait > wait_sec * 100) {
+				if (link_status == ODP_PKTIO_LINK_STATUS_UNKNOWN) {
+					printf("Warning (%s): Link status unknown after wait time.\n",
+					       name);
+					return 0;
+				}
+
+				ODPH_ERR("Error: (%s): Pktio link down after wait time.\n",
+					 name);
+				return -1;
```

Comment:
Maybe it would be better to print the error for all links that are still down, by using a break here instread of a return (and e.g. by setting ret = -1 and returning ret at the end)?